scribejava-apis/src/main/java/com/github/scribejava/apis/SalesforceApi.java

```diff
@@ -0,0 +1,72 @@
+package com.github.scribejava.apis;
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+
+import com.github.scribejava.apis.salesforce.SalesforceJsonTokenExtractor;
+
+/**
+ * This class is an implementation of the Salesforce OAuth2 API. 
+ */
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.extractors.TokenExtractor;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.Verb;
+
+public class SalesforceApi extends DefaultApi20 {
+	
+	// Constants
+	private static final String BASE_URL = "https://login.salesforce.com/services/oauth2";
+	private static final String ACCESS_URL = BASE_URL + "/token?grant_type=authorization_code";
+	private static final String AUTHORIZE_URL = BASE_URL + "/authorize";
+
+    protected SalesforceApi() {
+    	SSLContext sc;
+		try {
+			String protocol = SSLContext.getDefault().getProtocol();
+			if (!protocol.equals("TLSv1.2") || !protocol.equals("TLSv1.1")) { // This is important!
+				sc = SSLContext.getInstance("TLSv1.2");
+				sc.init(null, null, null);
+				HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());				
```

Comment:
Good question, will check.

Main reason for this "not that good" implementation is, that Scribe itself doesn't enforce or allow via  configuration on outgoing HTTPS connections the usage of TLS. So if a server admin hasn't changed the JVM defaults it'll use SSLv3. And that's bad.

Most providers disable SSLv3 nowadays and switch to a higher encryption layer. The alternative tp this specific implementation would be to allow the configuration of the to be used TLS layer via the scribe-core package.